src/fixup.js

```diff
@@ -139,4 +140,57 @@
     }
   }
 
+  /* Deprecation warning */
+  var request = new XMLHttpRequest();
+
+  request.open('GET', '//www.w3.org/TR/tr-outdated-spec');
+  request.onload = function() {
+    if (request.status < 200 && request.status > 400) {
+      return;
+    }
+    try {
+      var currentSpec = JSON.parse(request.responseText);
+    } catch (err) {
+      console.error(err);
+      return;
+    }
+    document.body.classList.add('outdated-spec');
+    var node = document.createElement("p");
+    node.classList.add("outdated-warning");
+
+    var warning =
+      "<strong>" +
+      currentSpec.header +
+      "</strong><span>" +
+      currentSpec.warning +
+      '<a id="outdated-note" href="' +
+      currentSpec.latestUrl +
+      '"> ' +
+      currentSpec.latestUrl +
+      '</a>.</span><input onclick="collapseWarning(false)" type="button" value="&#9662; collapse">';
+    node.innerHTML = warning;
+    document.body.appendChild(node);
+  };
+
+  request.onerror = function() {
+    console.error("Request to https://www.w3.org/TR/tr-outdated-spec failed.");
+  };
+
+  request.send();
 })();
+
+function collapseWarning(details) {
+  var node = document.querySelector(".outdated-warning")
+  ,   button = document.querySelector(".outdated-warning input");
+  if (details) {
+    node.classList.remove("outdated-collapsed");
+    button.value = '\u25BE collapse';
+    document.body.classList.add('outdated-spec');
+    button.onclick = function() {collapseWarning(false)};
```

Comment:
This continuous reassigning of the function seems really ugly. Why not just use a closure here? 

```JS
var isOpen = false;
var clickHandler = function() {
    isOpen = !isOpen;
   collapseWarning(isOpen);
}
button.onclick = clickHandler;
```

Reply:
Actually, it's also not great to do the query selection over and over... you can combine what I proposed above and get something much cleaner/maintainable:

```JS
var frag = document.createDocumentFragment();
var heading = document.createElement("strong");
heading.innerHTML = currentSpec.header;
frag.appendChild(heading);

var anchor = document.createElement("a");
anchor.id = "outdated-note";
anchor.href = currentSpec.latestUrl;
anchor.innerText = currentSpec.latestUrl + ".";
frag.appendChild(anchor);

var warning = document.createElement("<span>");
warning.innerText = currentSpec.warning;
warning.appendChild(anchor);
frag.appendChild(warning);

var button = document.createElement("button");
var handler = makeClickHandler(node);
button.addEventListener("click", handler);
button.innerHTML = "&#9662; collapse";
frag.appendChild(anchor);
node.appendChild(frag);

function makeClickHandler(node) {
  var isOpen = false;
  return function collapseWarning(event) {
    var button = event.target;
    isOpen = !isOpen;
    if (isOpen) {
      node.classList.remove("outdated-collapsed");
      button.innerText = "\u25BE collapse";
      document.body.classList.add("outdated-spec");
      return;
    }
    node.classList.add("outdated-collapsed");
    button.innerText = "\u25B4 expand";
    document.body.classList.remove("outdated-spec");
  };
}
```

Reply:
fixed typo in code above.

Reply:
realized that I don't need to pass in button, can just get it from the event.